src/modules/launcher/PowerLauncher/Helper/ThemeManager.cs

```diff
@@ -40,73 +41,70 @@ private void OnUserPreferenceChanged(object sender, UserPreferenceChangedEventAr
             {
                 string styleThemeString = theme switch
                 {
-                    ManagedCommon.Theme.Light => "Themes/Light.xaml",
-                    ManagedCommon.Theme.Dark => "Themes/Dark.xaml",
-                    ManagedCommon.Theme.HighContrastOne => "Themes/HighContrast1.xaml",
-                    ManagedCommon.Theme.HighContrastTwo => "Themes/HighContrast2.xaml",
-                    ManagedCommon.Theme.HighContrastWhite => "Themes/HighContrastWhite.xaml",
-                    _ => "Themes/HighContrastBlack.xaml",
+                    Theme.HighContrastOne => "Themes/HighContrast1.xaml",
+                    Theme.HighContrastTwo => "Themes/HighContrast2.xaml",
+                    Theme.HighContrastWhite => "Themes/HighContrastWhite.xaml",
+                    Theme.HighContrastBlack => "Themes/HighContrastBlack.xaml",
+                    _ => "Themes/Light.xaml",
                 };
+
                 _mainWindow.Resources.MergedDictionaries.Clear();
                 _mainWindow.Resources.MergedDictionaries.Add(new ResourceDictionary
                 {
                     Source = new Uri(styleThemeString, UriKind.Relative),
                 });
-                ResourceDictionary test = new ResourceDictionary
-                {
-                    Source = new Uri(styleThemeString, UriKind.Relative),
-                };
+
                 if (OSVersionHelper.IsWindows11())
                 {
                     // Apply background only on Windows 11 to keep the same style as WPFUI
                     _mainWindow.Background = new SolidColorBrush
                     {
-                        Color = (Color)_mainWindow.FindResource("LauncherBackgroundColor"), // Use your DynamicResource key here
+                        Color = (Color)_mainWindow.FindResource("LauncherBackgroundColor"),
                     };
                 }
             }
 
             ImageLoader.UpdateIconPath(theme);
-            ThemeChanged?.Invoke(_currentTheme, theme);
-            _currentTheme = theme;
+            ThemeChanged?.Invoke(CurrentTheme, theme);
+            CurrentTheme = theme;
         }
 
+        /// <summary>
+        /// Updates the application's theme based on the user's preference. If the Theme setting is
+        /// <see cref="Theme.System"/>, the current system theme is used; otherwise, the selected
+        /// theme is applied directly.
+        /// </summary>
         public void UpdateTheme()
         {
-            ManagedCommon.Theme newTheme = _settings.Theme;
-            ManagedCommon.Theme theme = ThemeExtensions.GetHighContrastBaseType();
-            if (theme != ManagedCommon.Theme.Light)
-            {
-                newTheme = theme;
-            }
-            else if (_settings.Theme == ManagedCommon.Theme.System)
-            {
-                newTheme = ThemeExtensions.GetCurrentTheme();
-            }
+            Theme newTheme = _settings.Theme == Theme.System ?
```

Comment:
There is an issue with this change.  In HC mode we load different background colors.
So if in Launcher settings you set either dark or light mode instead of System then in HC mode the colors will look differently. 
You can test this by running in any of the 4 HC mode in Windows 11 and then in launcher settings change the App Theme.

Reply:
Thank you for the review, @mantaionut

I can see that I need to immediately return the Contrast theme if one is active and matches one of the 4 defaults. I will update the code and add to the unit tests to cover the various cases.

However, even if I do that, I am not seeing the Light background applied if I change the setting to "Light" from "Windows default". I've checked both my new local build and an installed version of 0.88. It looks like there has been a change as part of the move to .NET 9, because 0.86 still works. I will raise this as a separate issue.

Reply:
Here's the separate Contrast theme issue: #37398

Reply:
@mantaionut  I have fixed the High Contrast priority issue and added unit tests to cover this in the latest commit. Please could you review?

The logic is now within `ThemeHelper.DetermineTheme`:

```csharp
public Theme DetermineTheme(Theme settingsTheme) =>
    GetHighContrastTheme() ??
        (settingsTheme == Theme.System ? GetAppsTheme() : ValidateTheme(settingsTheme));
```

This now always returns the HC theme, if one is active.

Relevant unit tests are in the `DetermineTheme_ReturnsExpectedTheme()` method in ThemeHelperTest.cs.

Thank you for your help.